async-raft/src/core/admin.rs

```diff
@@ -83,6 +83,7 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
     }
 
     #[tracing::instrument(level = "trace", skip(self, tx))]
+    #[allow(clippy::needless_collect)]
```

Comment:
Perhaps some recent changes are bringing this about. Perhaps down on `L109` we can remove the collect, and then just combine that code into the iterator chain which uses it down on line `L110`. Thoughts?

Reply:
I think this is an error, because it cannot be compiled without `collect`.  This will cause a warning when verifying the code through Clippy in Rust `v1.47`.

Reply:
Adding this annotation is just to avoid warnings, maybe you can find a way to remove this collect, which will be more efficient. I hope i can help you, but I need to understand your code, but it really has a certain degree of complexity.

Reply:
@sunli829 I am going to merge these changes, and then make a small modification to the code here (in attempt to reduce the number of clones). Once this is merged, I will get it prepped as a `v0.6.0` release for this crate.